lib/tools/adb-commands.js

```diff
@@ -275,56 +277,81 @@ methods.revokePermission = async function (pkg, permission) {
  *
  * @param {string} pkg - The package name to be processed.
  * @param {string} cmdOutput [null] - Optional parameter containing command output of
- *                                    _pm dump_ command. It speeds this method up
- *                                    if present.
+ *                                    _dumpsys package_ command. It may speed up the method execution.
  * @return {array} The list of granted permissions or an empty list.
  * @throws {Error} If there was an error while changing permissions.
  */
 methods.getGrantedPermissions = async function (pkg, cmdOutput = null) {
-  let stdout = cmdOutput || await this.shell(['pm', 'dump', pkg]);
-  let match = new RegExp(/(install permissions:|User 0)([\s\S]*?)DUMP OF SERVICE activity:/g).exec(stdout);
-  if (!match) {
-    throw new Error('Unable to get granted permissions');
+  const stdout = cmdOutput || await this.shell(['dumpsys', 'package', pkg]);
+  const grantedPermissions = [];
+  for (const pattern of [
+    new RegExp(/^(\s*install permissions:[\s\S]+)/, 'm'),
+    new RegExp(/^(\s*runtime permissions:[\s\S]+)/, 'm'),
+  ]) {
+    const match = pattern.exec(stdout);
+    if (!match) {
+      continue;
+    }
+
+    grantedPermissions.push(
+      ...(extractPermissionsFromGroup(match[1])
+        .filter((item) => item.granted === true)
+        .map((item) => item.permission)));
+  }
+  if (_.isEmpty(grantedPermissions)) {
+    log.info(`No permissions were granted for '${pkg}'`);
   }
-  return (match[0].match(/android\.permission\.\w+:\sgranted=true/g) || [])
-    .map((x) => x.replace(/:\sgranted=true/g, ''));
+  return grantedPermissions;
 };
 
 /**
  * Retrieve the list of denied permissions for the particular package.
  *
  * @param {string} pkg - The package name to be processed.
  * @param {string} cmdOutput [null] - Optional parameter containing command output of
- *                                    _pm dump_ command. It speeds this method up
- *                                    if present.
+ *                                    _dumpsys package_ command. It may speed up the method execution.
  * @return {array} The list of denied permissions or an empty list.
  */
 methods.getDeniedPermissions = async function (pkg, cmdOutput = null) {
-  let stdout = cmdOutput || await this.shell(['pm', 'dump', pkg]);
-  let match = new RegExp(/(install permissions:|User 0)([\s\S]*?)DUMP OF SERVICE activity:/g).exec(stdout);
-  if (!match) {
-    throw new Error('Unable to get denied permissions');
+  const stdout = cmdOutput || await this.shell(['dumpsys', 'package', pkg]);
```

Comment:
this is all the same logic as above; maybe factor out one method that just takes a bool to get granted vs denied?

Reply:
I don't want to change module interface. And there is also not much content to extract into a common function